jetty-webapp/src/test/java/org/eclipse/jetty/webapp/TempDirTest.java

```diff
@@ -0,0 +1,266 @@
+        assertThrows(IllegalStateException.class, () -> webInfConfiguration.resolveTempDirectory(webAppContext));
+    }
+
+    /**
+     * Test ServletContext.TEMPDIR as valid directory with types File, String and Path.
+     */
+    @ParameterizedTest
+    @ValueSource(strings = {"File", "String", "Path"})
+    public void attributeWithValidDirectory(String type) throws Exception
+    {
+        WebAppContext webAppContext = new WebAppContext();
+        Path tmpDir = Files.createTempDirectory("jetty_test");
+        switch (type)
+        {
+            case "File":
+                webAppContext.setAttribute(ServletContext.TEMPDIR, tmpDir.toFile());
+                break;
+            case "String":
+                webAppContext.setAttribute(ServletContext.TEMPDIR, tmpDir.toString());
+                break;
+            case "Path":
+                webAppContext.setAttribute(ServletContext.TEMPDIR, tmpDir);
+                break;
+            default:
+                throw new IllegalStateException();
+        }
+
+        // Test we have correct value as the webapp temp directory.
+        WebInfConfiguration webInfConfiguration = new WebInfConfiguration();
+        webInfConfiguration.resolveTempDirectory(webAppContext);
+        assertThat(webAppContext.getTempDirectory().toPath(), is(tmpDir));
+    }
+
+    /**
+     * ServletContext.TEMPDIR as File to a non existent directory.
+     */
+    @ParameterizedTest
+    @ValueSource(strings = {"File", "String", "Path"})
+    public void attributeWithNonExistentDirectory(String type) throws Exception
+    {
+        WebAppContext webAppContext = new WebAppContext();
+        Path tmpDir = Files.createTempDirectory("jetty_test").resolve("foo_test_tmp");
+        Files.deleteIfExists(tmpDir);
+        assertFalse(Files.exists(tmpDir));
+        switch (type)
+        {
+            case "File":
+                webAppContext.setAttribute(ServletContext.TEMPDIR, tmpDir.toFile());
+                break;
+            case "String":
+                webAppContext.setAttribute(ServletContext.TEMPDIR, tmpDir.toString());
+                break;
+            case "Path":
+                webAppContext.setAttribute(ServletContext.TEMPDIR, tmpDir);
+                break;
+            default:
+                throw new IllegalStateException();
+        }
+
+        // Test we have correct value as the webapp temp directory.
+        WebInfConfiguration webInfConfiguration = new WebInfConfiguration();
+        webInfConfiguration.resolveTempDirectory(webAppContext);
+        Path webappTmpDir = webAppContext.getTempDirectory().toPath();
+        assertThat(webappTmpDir, is(tmpDir));
+        assertTrue(Files.exists(webappTmpDir));
+    }
+
+    /**
+     * ServletContext.TEMPDIR has invalid <code>String</code> directory value
+     * IllegalStateException
+     */
+    @Disabled
+    @Test
+    public void attributeWithInvalidStringValue() throws Exception
```

Comment:
@olamy how is this different to the `attributeWithNonExistentDirectory` test?

I don't think `/French/Cheese/Rocks` is  an invalid value, just a directory which doesn't exist. This could be why it was failing CI but passes for me locally, because in CI we are running as root so we do have permission to create this.

Reply:
yup the test fail on CI because it is executed as root so as root the random value directory `/French/Cheese/Rocks` can be created but the problem is I can't run the docker images as non root otherwise all the testcontainers will not works.
both tests  `attributeWithInvalidPermissionStringValue` and `attributeWithInvalidStringValue` do almost the same except one is using a totally random value whereas this other one try to create a directory within an existing directory `/var` which is not writable for non root. 
The test attributeWithNonExistentDirectory is different because it create a directory within the java.io.tmpdir so every users can do it even non root.
I don't mind having those tests disabled until I find a way to run them as non root in Jenkins but alas cannot find a way yet.